experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,202 @@
+  return sum((i-1)*l_i for (i, l_i) in enumerate(lambda); init=0)
+end
+
+# Computes the b-invariant of a multipartition
+function b_inv(lbb::Multipartition)
+  return b1(map(lambda -> sum(lambda)[1], lbb)) + length(lbb)*sum(i -> b1(data(lbb[i])),1:length(lbb))
+end
+
+function beta_number_to_partition(beta::Vector{Int})
+  lb=Int[]
+  lbeta=length(beta)
+  for j in 1:lbeta
+    nbholes=beta[j]-beta[1]-(j-1)
+    if nbholes >= 1
+    pushfirst!(lb, nbholes)
+    end
+  end
+  return partition(lb)
+end
+
+# Residue notation is in terms of rows and columns
+
+function core(coroot::Vector{Int},r::Int)
+  beta=Int[]
+  m=minimum(coroot)
+  M=maximum(coroot)
+  for k in m:M
+    for j in 1:r
+       if k <= coroot[j]
+          push!(beta, k*r+(j-1))
+       end
+    end
+  end
+  beta=sort!(beta)
+  return beta_number_to_partition(beta)
+end
+
+#Inspired by Sagemath's algorithm to obtain a partition from core and quotient
+
+function tau_om(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int})
+   r=length(lbb)
+   w0=perm([r-i for i in 0:(r-1)])
+   wperm=inv(wperm)
+   lbb_perm=multipartition(permuted(lbb.mp,wperm*w0))
+   gamma=core([-x for x in coroot],r)
+   lg=length(gamma)
+   k=r*maximum(length(lbb_perm[i]) for i in 1:r) + lg
+   v=[[gamma[i]-(i-1) for i in 1:lg]; [-i for i in lg:(k-1)]]
+   w=[[x for x in v if mod((x-i),r) == 0] for i in 1:r]
+   new_w=[]
+   for i in 1:r
+     lw=length(w[i])
+     lq=length(lbb_perm[i])
+     new_w=[new_w;[w[i][j] + r*lbb_perm[i][j] for j in 1:lq]]
+     new_w=[new_w;[w[i][j] for j in lq+1:lw]]
+   end
+   sort!(new_w,rev=true)
+   new_w=[new_w[i]+(i-1) for i in 1:length(new_w)]
+   filter!(x-> x!=0,new_w)
+   return partition(new_w)
+end
+
+#Coroots are given in the canonical basis (e_i) of Z^I.
+#Note also that \alpha^{\vee}_i=e_{i-1}-e_i.
+#The coroot lattice is thus the lattice of null sum elements in Z^I.
+
+#Reoders the CharTable from Chevie
+function reorder(charTirr::Matrix{QQAbFieldElem{AbsSimpleNumFieldElem}}, r::Int, n::Int, Modules::Vector{Multipartition{Int}}, K::QQAbField{AbsSimpleNumField})
+  mps=multipartitions(n,r)
+  new_ord=Int[]
+    for lbb in mps
+      i=findfirst(x-> x==lbb,Modules)
+      push!(new_ord,i)
+    end
+  return charTirr[new_ord,:]
+end
+
+#Tools from representation theory
+
+# NEED TO FIX THAT (issue is with the fraction_field and canonical_unit...)
+# Computes the fake degree of a multipartition cf. Ste89 Thm 5.3 and Prop. 3.3.2 Haiman cdm
+function fake_deg(lbb::Multipartition, K::QQAbField{AbsSimpleNumField}, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}})
+  r=size(lbb)[1]
+  n=sum(lbb)[1]
+  res=Q(1)
+  for p in 1:n
+    res=res*(1-var^(r*p))
+  end
+  for k in 1:r
+    for i in 1:length(lbb[k])
+      for j in 1:lbb[k][i]
+        hookfactor=(Q(1)-var^(r*(1+lbb[k][i]+length(findall(row->row >= j,lbb[k]))-j-i)))
```

Comment:
```suggestion
        hookfactor=(Q(1)-var^(r*(1+lbb[k][i]+count(>=(j) ,lbb[k]))-j-i))
```
`count(predicate, iterable)` should do the same as `length(findall(predicate, iterable))`, but in a more efficient manner